### PR TITLE
Delegation sc integration tests

### PR DESCRIPTION
**Pull request by BeniaminDrasovean**

### CODE REVIEW

File: integrationTests/vm/delegation/delegationScenarios_test.go [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract"
+	"github.com/ElrondNetwork/elrond-go/vm"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDelegationSystemNodesOperations(t *testing.T) {
```
Comment (iulianpascalau) [inline integrationTests/vm/delegation/delegationScenarios_test.go:26]: :hatoff: this would have taken at least 5 minutes to wait until completion. In this implementation it only took 40ms

File: process/block/postprocess/testIntermediateResult.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+// TestIntermediateResProc --
```
Comment (iulianpascalau) [inline process/block/postprocess/testIntermediateResult.go:13]: Please add in the comment that this should only be used for testing purposes not in production code. Example: 
https://github.com/ElrondNetwork/elrond-go/blob/master/process/economics/testEconomicsData.go

File: process/block/postprocess/testIntermediateResult.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+type TestIntermediateResProc struct {
+	*intermediateResultsProcessor
+}
+
+// NewTestIntermediateResultsProcessor --
```
Comment (iulianpascalau) [inline process/block/postprocess/testIntermediateResult.go:18]: Might add some real comments here

File: vm/systemSmartContracts/delegation.go [hunk 4]
```diff
@@ -1430,6 +1456,307 @@ func (d *delegation) withdraw(args *vmcommon.ContractCallInput) vmcommon.ReturnC
 	return vmcommon.Ok
 }
 
+func (d *delegation) checkArgumentsForGeneralViewFunc(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/delegation.go:1459]: :+1:  we can make this part of a common "something" that will be embedded in each system SC

File: vm/systemSmartContracts/delegation.go [hunk 5]
```diff
@@ -1430,6 +1456,307 @@ func (d *delegation) withdraw(args *vmcommon.ContractCallInput) vmcommon.ReturnC
...
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+
+	d.eei.Finish(globalFund.TotalUnBondedFromNodes.Bytes())
+	return vmcommon.Ok
+}
+
+func (d *delegation) checkArgumentsForUserViewFunc(args *vmcommon.ContractCallInput) (*DelegatorData, vmcommon.ReturnCode) {
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/delegation.go:1656]: this can also be move to that common "something"

File: vm/systemSmartContracts/delegation.go (hunk 4 above)
Comment (iulianpascalau) [inline vm/systemSmartContracts/delegation.go:1459]: we need unit tests for these view functions (may be another PR)

**Review by iulianpascalau — APPROVED** [review]
:+1: GJ!

**Review by sasurobert — APPROVED** [review]
